### PR TITLE
feat(history): diff schemas to get unknown attributes

### PR DESCRIPTION
**Pull request by markkaylor**
### What does it do?

- Add a meta object to `findVersionsPage` containing unknown attributes 
- Refactor tests into describes

### Why is it needed?

- So we can determine what can and cannot be restored

### How to test it?

- Go to a contentType schema and add an attribute like `newField`
- Make a request to findVersionsPage
- You should have a `meta.unknownAttributes` object with:
```
{
	added: {
	  newField: {type: 'string'}
	}
}
```
- Remove an attribute and you should have
```
{
	added: {
	  newField: {type: 'string'}
	},
	removed: {
	  fieldYouRemoved: {type: 'string'}
	},
}
``` 
- Rename an attribute and you should have
```
{
	added: {
	  newField: {type: 'string'}
      renamedField: {type: 'string'}
	},
	removed: {
	  fieldYouRemoved: {type: 'string'}
      fieldYouRenamed: : {type: 'string'}
	},
}
```

### CODE REVIEW

File: packages/core/content-manager/server/src/history/services/history.ts [hunk 1]
```diff
@@ -183,7 +226,16 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
...
 
-      const sanitizedResults = results.map((result) => ({
+      const versionsWithMeta = results.map((version) => {
+        return {
+          ...version,
+          meta: {
+            unknownAttributes: getAttributesDiff(version.schema, version.contentType),
+          },
+        };
+      });
```
Comment (markkaylor) [inline packages/core/content-manager/server/src/history/services/history.ts:236]: ℹ️ Hesitated to do this here vs in the controller 🤔

File: packages/core/content-manager/server/src/history/services/__tests__/history.test.ts [hunk 2]
```diff
@@ -95,136 +122,162 @@ describe('history-version service', () => {
...
+          added: {
+            newField: { type: 'string' },
+            renamed: { type: 'string' },
+          },
+          removed: {
+            title: { type: 'string' },
+          },
+        },
+      });
+    });
```
Comment (markkaylor) [inline packages/core/content-manager/server/src/history/services/__tests__/history.test.ts:281]: ℹ️ This is the only test I added and associated mocks. Sorry should have done a separate commit for organizing everything into describes.

File: packages/core/content-manager/server/src/history/services/history.ts [hunk 3]
```diff
@@ -59,6 +70,49 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
     return documentMetadataService.getStatus(document, meta.availableStatus);
   };
 
+  /**
+   * Get the difference between the version schema and the content type schema
+   * Returns the attributes with their original shape
+   */
+  const getAttributesDiff = (
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/history/services/history.ts:77]: fyi, there is a utility to diff schemas in `packages/core/database/src/schema/diff.ts` , it doesn't look like it's exported though. I have still not looked at the details but it's kind of accessible through `strapi.db.schema.schemaDiff.diff()` , if you are interested we could sync with alex to check if it's ok to use that here

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/history/services/history.ts:77]: code is not long, so I don't think its a bad idea to keep your implementation either. Mentioned just in case!

Comment (markkaylor) [reply packages/core/content-manager/server/src/history/services/history.ts:77]: 🤦‍♂️ didn't even think to look if something already existed. I'll take a look thanks.

**Review by Marc-Roig — APPROVED** [review]

File: packages/core/content-manager/server/src/history/services/history.ts (hunk 3 above)
Comment (markkaylor) [reply packages/core/content-manager/server/src/history/services/history.ts]: I took a look but it does look like it is diffing database columns so I don't think it is applicable here.

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/history/services/history.ts]: 🫡 makes sense
